packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderApiKeyAuthTests.cs

```diff
@@ -0,0 +1,433 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Generator.CSharp.ClientModel.Providers;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using Microsoft.Generator.CSharp.Snippets;
+using Microsoft.Generator.CSharp.Statements;
+using Microsoft.Generator.CSharp.Tests.Common;
+using NUnit.Framework;
+using static Microsoft.Generator.CSharp.Snippets.Snippet;
+
+namespace Microsoft.Generator.CSharp.ClientModel.Tests.Providers.ClientProviders
+{
+    public class ClientProviderApiKeyAuthTests
+    {
+        private const string SubClientsCategory = "WithSubClients";
+        private const string TestClientName = "TestClient";
+        private static readonly InputClient _animalClient = new("animal", "AnimalClient description", [], [], TestClientName);
+        private static readonly InputClient _dogClient = new("dog", "DogClient description", [], [], _animalClient.Name);
+        private static readonly InputClient _huskyClient = new("husky", "HuskyClient description", [], [], _dogClient.Name);
+
+        [SetUp]
+        public void SetUp()
+        {
+            var categories = TestContext.CurrentContext.Test?.Properties["Category"];
+            bool containsSubClients = categories?.Contains(SubClientsCategory) ?? false;
+
+            if (containsSubClients)
+            {
+                MockHelpers.LoadMockPlugin(
+                    apiKeyAuth: () => new InputApiKeyAuth("mock", null),
+                    clients: () => [_animalClient, _dogClient, _huskyClient]);
+            }
+            else
+            {
+                MockHelpers.LoadMockPlugin(apiKeyAuth: () => new InputApiKeyAuth("mock", null));
+            }
+        }
+
+        [Test]
+        public void TestBuildProperties()
```

Comment:
How is this test specific to auth?

Reply:
This has been moved back to `ClientProviderTests`